src/sage/libs/ntl/ntl_ZZ_pX.pyx

```diff
@@ -501,16 +502,67 @@ cdef class ntl_ZZ_pX():
             sage: g = ntl.ZZ_pX([-1,0,1],c)
             sage: g**10
             [1 0 10 0 5 0 0 0 10 0 8 0 10 0 0 0 5 0 10 0 1]
+
+            sage: x = ntl.ZZ_pX([0,1],c)
+            sage: x**10
+            [0 0 0 0 0 0 0 0 0 1]
         """
+        if modulus is None:
+            return self._pow(n)
+        else:
+            return self._powmod(n, modulus)
+
+    def _pow(ntl_ZZ_pX self, n):
```

Comment:
Why not make this into a (inline) `cdef` function? It is much faster to call.

Reply:
Changed to a `cpdef` (it is quite unlikely to make a noticeable difference since `pow` is a costly operation)

Reply:
Thanks. Most likely, but it might matter in small cases.